pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
What extra args to bash do you envision people wanting to pass?
I saw the point with mvn, but with bash -c I  normally pass nothing else.

The name "Shell" indeed needs to change, I'll do that tomorrow.

Can we use /bin/sh instead of the original proposal of /bin/bash? I
realized that more minimalist images like busybox have the former but not
the later.
On Feb 1, 2016 9:43 PM, "Clayton Coleman" <notifications@github.com> wrote:

> In pkg/build/api/v1beta3/types.go
> <https://github.com/openshift/origin/pull/6715#discussion_r51477432>:
>
> > @@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
> >  	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
> >  }
> >
> > +// BuildPostCommitSpec holds a build post commit hook specification.
> > +type BuildPostCommitSpec struct {
> > +	// Shell is an action that executes a shell script in a container
> > +	// running the build output image.
> > +	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
>
> The positional args thing I think is what gets me over the hump - it's the
> closest to what I want and it's supported in bash - /bin/bash -c STRING
> -- ARG0 ARG1, passed as an array of argument into the container launch
> command (so that we preserve strong argument separation).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6715/files#r51477432>.
>